app/src/main/java/com/kinkong/database/FBDatabase.java

```diff
@@ -43,7 +42,7 @@ public void cacheBasicData(final ValueEventListener valueEventListener) {
             @Override
             public void onDataChange(DataSnapshot dataSnapshot) {
                 if (dataSnapshot.getValue() != null) {
-                    nextQuestionNum = dataSnapshot.getValue(Integer.class);
+                    nextQuestionIndexOrigin = nextQuestionNum = dataSnapshot.getValue(Integer.class);
```

Comment:
I think double assignments make code harder to understand.
Following would be more readable:
```java
firstQuestionIndex = dataSnapshot.getValue(Integer.class);
nextQuestionIndex = firstQuestionIndex;
```